source/Cosmos.System2/Network/IPV4/UDP/DHCP/DHCPClient.cs

```diff
@@ -0,0 +1,228 @@
+        /// <returns></returns>
+        public static Address DHCPServerAddress(NetworkDevice networkDevice)
+        {
+            return NetworkConfig.Get(networkDevice).DefaultGateway;
+        }
+
+        /// <summary>
+        /// Create new inctanse of the <see cref="UdpClient"/> class.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown on fatal error (contact support).</exception>
+        /// <exception cref="ArgumentException">Thrown if UdpClient with localPort 0 exists.</exception>
+        public DHCPClient()
+        {
+            rxBuffer = new Queue<DHCPPacket>(8);
+            currentClient = this;
+        }
+
+        /// <summary>
+        /// Close connection.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown on fatal error (contact support).</exception>
+        public void Close()
+        {
+            currentClient = null;
+        }
+
+        /// <summary>
+        /// Receive data from packet.
+        /// </summary>
+        /// <param name="packet">Packet to receive.</param>
+        /// <exception cref="OverflowException">Thrown on fatal error (contact support).</exception>
+        /// <exception cref="Sys.IO.IOException">Thrown on IO error.</exception>
+        public void receiveData(DHCPPacket packet)
+        {
+            rxBuffer.Enqueue(packet);
+        }
+
+        /// <summary>
+        /// Receive data
+        /// </summary>
+        /// <param name="timeout">timeout value, default 5000ms
+        /// <returns>time value (-1 = timeout)</returns>
+        /// <exception cref="InvalidOperationException">Thrown on fatal error (contact support).</exception>
+        private int Receive(int timeout = 5000)
+        {
+            int second = 0;
+            int _deltaT = 0;
+
+            while (rxBuffer.Count < 1)
+            {
+                if (second > (timeout / 1000))
+                {
+                    return -1;
+                }
+                if (_deltaT != Cosmos.HAL.RTC.Second)
+                {
+                    second++;
+                    _deltaT = Cosmos.HAL.RTC.Second;
+                }
+            }
+
+            var packet = rxBuffer.Dequeue();
+
+            if (packet.MessageType == 2) //Boot Reply
+            {
+                if (packet.RawData[284] == 0x02) //Offer packet received
+                {
+                    SendRequestPacket(packet.Client, packet.Server);
+                }
+                else if (packet.RawData[284] == 0x05 || packet.RawData[284] == 0x06) //ACK or NAK DHCP packet received
+                {
+                    DHCPAck ack = new DHCPAck(packet.RawData);
+                    if (asked)
+                    {
+                        Apply(ack, true);
+                    }
+                    else
+                    {
+                        Apply(ack);
+                    }
+                }
+            }
+
+            return second;
```

Comment:
Is it important to return how much time was taken? Would it not be enough to just return -1 or false when it fails and 0 or true when it receives something?